js/validate.js

```diff
@@ -173,17 +175,31 @@ function validate_isPasswd(s)
 	return (s.length >= 5 && s.length < 255);
 }
 
-$(document).on('focusout', 'input.validate, textarea.validate', function() {
-	if ($(this).hasClass('is_required') || $(this).val().length)
+function validate_field(that)
+{
+	if ($(that).hasClass('is_required') || $(that).val().length)
 	{
-		if ($(this).attr('name') == 'postcode' && typeof(countriesNeedZipCode[$('#id_country option:selected').val()]) != 'undefined')
-			var result = window['validate_'+$(this).attr('data-validate')]($(this).val(), countriesNeedZipCode[$('#id_country option:selected').val()]);
-		else
-			var result = window['validate_'+$(this).attr('data-validate')]($(this).val())
+		if ($(that).attr('data-validate') == 'isPostCode')
+		{
+			var selector = '#id_country';
+			if ($(that).attr('name') == 'postcode_invoice')
+				selector += '_invoice';
+
+			var id_country = $(selector + ' option:selected').val();
+
+			if (typeof(countriesNeedZipCode[id_country]) != 'undefined' && typeof(countries[id_country]['iso_code']) != 'undefined')
```

Comment:
If countriesNeedZipCode[id_country] is defined but countries[id_country]['iso_code'] is undefined, validate_isPostCode never gets executed, but line 122 is already prepared for an undefined iso_code